### ISSUE TITLE
`no-unbound-method` errors even for non-problematic references

### ISSUE THREAD

**Issue by ethanresnick** [palantir/tslint#2128]
### Bug Report

- __TSLint version__: `4.4.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: cli

#### TypeScript code being linted

```ts
// Check if an optional method exists on obj (or its prototype chain)
function a(obj) {
  return obj && obj.method !== undefined;
}
```

with `tslint.json` configuration:

```json
{"no-unbound-method": true}
```

#### Actual behavior
no-unbound-method gives an error: `Avoid referencing unbound methods which may cause unintentional scoping of 'this'`.

#### Expected behavior

Storing a reference to the unbound method should give an error, but simply referencing it without capturing that reference should not.

**Comment by rhysd** [comment]
I think `.apply` and `.call` are also exceptional case for `no-unbound-method` rule because they binds value to `this`.

```typescript
const a = [0];
Array.prototype.push.apply(a, [1]); // This line causes an error
console.log(a);
```

**Comment by nchen63** [comment]
cc: @andy-hanson

**Comment by nchen63** [comment]
fixed
